app/src/main/java/org/dhis2/usescases/splash/SplashPresenter.kt

```diff
@@ -1,63 +1,55 @@
 package org.dhis2.usescases.splash
 
 
-import android.content.Context
-import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
-import io.reactivex.schedulers.Schedulers
+import org.dhis2.Bindings.parseToList
 import org.dhis2.data.prefs.Preference
+import org.dhis2.data.prefs.PreferenceProvider
 import org.dhis2.data.schedulers.SchedulerProvider
 import org.dhis2.data.server.UserManager
-import org.dhis2.usescases.login.LoginActivity
-import org.dhis2.usescases.main.MainActivity
-import org.dhis2.usescases.sync.SyncActivity
 import org.dhis2.utils.Constants
 import timber.log.Timber
+import java.util.*
 import java.util.concurrent.TimeUnit
 
-class SplashPresenter internal constructor(private val userManager: UserManager?, private val schedulerProvider: SchedulerProvider) : SplashContracts.Presenter {
-    private var view: SplashContracts.View? = null
+class SplashPresenter internal constructor(
+        private var view: SplashView,
+        private val userManager: UserManager?,
+        private val schedulerProvider: SchedulerProvider,
+        private val preferenceProvider: PreferenceProvider) {
+
     var compositeDisposable: CompositeDisposable = CompositeDisposable()
 
-    override fun destroy() {
+    fun destroy() {
         compositeDisposable.clear()
     }
 
-    override fun init(view: SplashContracts.View) {
-        this.view = view
+    fun init() {
+        preferenceProvider.sharedPreferences().all.forEach {
```

Comment:
For next time :+1:  I would move this code to preferences provider o a class with only the responsability of migration one preferences to another. we should try to have presenters only with the "view logic" and fetching from data